modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/MetadataComponentModelValidator.java

```diff
@@ -181,8 +183,8 @@ private void validateMetadataReturnType(ExtensionModel extensionModel, Component
 
         @Override
         public void visitObject(ObjectType objectType) {
-          objectType.getOpenRestriction().ifPresent(t -> failIfTypeIsObject(component, extensionModel, t, problemsReporter));
-          failIfTypeIsObject(component, extensionModel, objectType, problemsReporter);
+          objectType.getOpenRestriction().ifPresent(t -> checkValidType(component, extensionModel, t, problemsReporter));
+          checkValidType(component, extensionModel, objectType, problemsReporter);
```

Comment:
is this ok? you will execute this even if the open restriction is present. Shouldn't it be one or the other?